File: totient.cpp
#include <iostream>

bool check(int number, int divisor){
    if (number % divisor == 0){
        return true;
    } else {
        return false;
    }
    
    // You can replace the entire function body with this if you want, though
    // this is less obvious.
    //return !(number % divisor)
}

bool isRelativePrime(int num1, int num2){
    for (int divisor = 2; divisor < num1 + 1; divisor++){
        // Checking if the result is 1 is uneccesary, you can simply put it in the if statement
        if (check(num1, divisor)){
            // Checking if the result is 1 is uneccesary, you can simply put it in the if statement
            if (check(num2, divisor)){
                return false;
            }
        }
    }
    return true;
}

int totient(int num1){
    int count = 0;
    for (int num2 = 1; num2 < num1 + 1; num2++){
        // Checking if the result is 1 is uneccesary, you can simply put it in the if statement
        if (isRelativePrime(num1,num2)){
            count++;
        }
    }
    return count;
}

int main(){
    for (int i = 1; i < 501; i++){
        std::cout << "φ(" << i << ") = " << totient(i) << std::endl;
    }
    return 0;
}

File: totient.py
from functools import cache
from functools import lru_cache
#import iostream :joy_cat:

@cache
def check(number, divisor):
    return True if number % divisor == 0 else False

@lru_cache(maxsize=6)
def isRelativePrime(num1, num2):
    for divisor in range(2,num1 + 1):
        if check(num1, divisor):
            if check(num2, divisor):
                return False
    return True

@lru_cache(maxsize=20)
def totient(num1):
    count = 0
    for num2 in range(1,num1 + 1):
        if isRelativePrime(num1,num2):
            count += 1
    return count

if __name__ == "__main__":
    for i in range(501):
        print(f"φ({i}) = {totient(i)}")